### PR TITLE
Render prompt on precmd, prefix functions with prompt_geometry

### PR DESCRIPTION
**Pull request by jedahan**
The main fix is to create a setup function that runs prompt_geometry() on precmd. Not sure if its the right way to do things, but its what works for me now.

### CODE REVIEW

**Comment by jedahan** [comment]
Should fix #4

**Comment by frm** [comment]
Thanks for this!

The issue with #4 was using single quotes instead of double quotes, I'm assuming?

**Comment by jedahan** [comment]
The quotes were one issue, but then I found another issue - the prompt was only being rendered one time on startup, not after every command. If you'd like, I can split this pull request into 3 commits that should make things more clear:
- rename private functions to have `prompt_geometry_` prefix
- switch from single to double quote `' => "`
- render prompt on precmd or preexec

**Comment by frm** [comment]
If you could split it for clarity it would be awesome!

File: geometry.zsh [hunk 1]
```diff
@@ -112,17 +112,20 @@ prompt_geometry_set_title() {
...
   PROMPT2=" $RPROMPT_SYMBOL "
   RPROMPT="$(prompt_geometry_git_info)"
 }
 
+prompt_geometry_setup() {
+  autoload -U add-zsh-hook
+
+  add-zsh-hook preexec prompt_geometry_set_cmd_title
+  add-zsh-hook precmd prompt_geometry_set_title
+  add-zsh-hook precmd prompt_geometry_render
```
Comment (frm) [inline geometry.zsh:128]: Nice change. I noticed the first version was using the `set_title` function to re-render the prompt, which wasn't clear at all, this should be the correct way. 👍

**Pull request merged**